### PR TITLE
ci: add check-deprecated-file-change.yaml

### PR DESCRIPTION
**Pull request by kenji-miyake**
To prevent contributors from editing `cspell/.cspell.json`.

### CODE REVIEW

**Comment by kenji-miyake** [comment]
No change to `.cspell.json`.
https://github.com/tier4/autoware-spell-check-dict/runs/5828491277?check_suite_focus=true
-> Passed as expected.

Edit `.cspell.json`.
https://github.com/tier4/autoware-spell-check-dict/runs/5828505334?check_suite_focus=true
-> Passed as expected.

Edit `cspell/.cspell.json`.
https://github.com/tier4/autoware-spell-check-dict/runs/5828515653?check_suite_focus=true
-> Failed as expected.

File: .github/workflows/check-deprecated-file-change.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check deprecated file change
+        run: |
+          if ! git diff --exit-code --quiet "origin/${{ github.base_ref }}...HEAD" -- cspell/.cspell.json; then
+            echo "Don't edit 'cspell/.cspell.json', edit '.cspell.json' instead."
```
Comment (KeisukeShima) [inline .github/workflows/check-deprecated-file-change.yaml:18]: How about adding a warning message?
```suggestion
            echo "::warning file=cspell/.cspell.json::Don't edit 'cspell/.cspell.json', edit '.cspell.json' instead."
```

Comment (kenji-miyake) [reply .github/workflows/check-deprecated-file-change.yaml:18]: Thank you! It's nice. :+1:

**Review by KeisukeShima — APPROVED** [review]
LGTM :100:

**Pull request merged**